src/CommandingOfficers/MeteorParams.java

```diff
@@ -0,0 +1,116 @@
+package CommandingOfficers;
+
+import java.util.ArrayList;
+
+import Engine.Utils;
+import Engine.XYCoord;
+import Engine.GameEvents.MassDamageEvent;
+import Terrain.GameMap;
+import Terrain.MapMaster;
+import Units.Unit;
+
+public class MeteorParams
+{
+  public XYCoord target; // This is allowed to be null. If it is, then this class will produce a no-op MassDamageEvent.
+  public int power = 3; // Silo
+  public int radius = 2; // ditto
+  public boolean selfHarm = true;
+  public boolean inflictStun = false;
+  public MeteorParams(XYCoord targetPosition, int radius)
+  {
+    target = targetPosition;
+    this.radius = radius;
+  }
+
+  public MassDamageEvent getDamage(MapMaster map, Commander attacker)
+  {
+    if( null == target )
+      return new MassDamageEvent(attacker, new ArrayList<>(), power*10, false, inflictStun);
+
+    ArrayList<XYCoord> locations = Utils.findLocationsInRange(map, target, 0, radius);
+    ArrayList<Unit> victimList = new ArrayList<>();
+    for( XYCoord loc : locations )
+    {
+      Unit victim = map.getResident(loc);
+      if( victim != null && // Something is there.
+          (selfHarm || victim.CO.isEnemy(attacker)) ) // It's not friendly enough to spare.
+      {
+        victimList.add(victim);
+      }
+    }
+    return new MassDamageEvent(attacker, victimList, power*10, false, inflictStun);
+  }
+
+  /**
+   * @return The aggregate score for the tile based on all known victims.
+   */
+  private static int getScore(GameMap map, Commander attacker, XYCoord target, int radius, IValueFinder evaluator)
+  {
+    int currentValue = 0;
+
+    ArrayList<XYCoord> locations = Utils.findLocationsInRange(map, target, 0, radius);
+    for( XYCoord loc : locations )
+    {
+      Unit found = map.getResident(loc);
+      if( found != null ) // Something is there.
+      {
+        currentValue += evaluator.getValue(attacker, found);
+      }
+    }
+
+    return currentValue;
+  }
+
+  /**
+   * @return A MassStrikeParams with the best target, or no target if there are no values above 0.
```

Comment:
Return type is wrong. Same for `findValueOnEnemy` below.

Also these names are not... ideal. `getScore` and `findValue` just sound like synonyms, and `findValueOnEnemy` is just confusing.

Reply:
So now we can plan meteor strikes against the enemy specifically, or just plan them? Is the difference just in the use of two maps to make AW2 Sturm work right? Is one a special case of the other? I could try to read the code and find out, but it's a weeknight.

Reply:
"On enemy" means "centered on a particular enemy unit", so it is indeed a special case of the other one. I guess the docstring could say that.
One map is used to check whether an enemy is on the center tile (so you can hit there), the other is used to calculate the expected value of the strike.

I thought about making that logic live in the value finder interface, but figured it fit better this way.

Reply:
> "On enemy" means "centered on a particular enemy unit"

Ah, I didn't make that connection. Just looked like it meant "at the enemy" which seems... obvious?

> EDIT: Past me agreed

So you did. Must have glossed over it, maybe because the comment on the second just says it restricts targeting, and doesn't otherwise connect itself to the first (I mean the names are similar, but they also aren't overloads). Code monkey like to be hit in the face with extremely unsubtle comments, like "This function is the same as `planMeteor`, except it restricts targeting...", and then probably put a parallel/reversed comment on the other one stating that it doesn't have the restriction.

> it is indeed a special case of the other one

Could there just be an optional Boolean to restrict targeting? I guess that might be cumbersome given the map treatment differences. Probably fine as is, I'm sure you considered consolidating things.

Reply:
> Ah, I didn't make that connection. Just looked like it meant "at the enemy" which seems... obvious?

Yeah, "On" isn't the most disambiguated word.
> Code monkey like to be hit in the face with extremely unsubtle comments

I suppose that's fair.
> Could there just be an optional Boolean to restrict targeting? I guess that might be cumbersome given the map treatment differences. Probably fine as is, I'm sure you considered consolidating things.

Not super conveniently.
The "on" variant needs a separate map for finding units to center on, which isn't needed at all on the other version.

I was also originally planning to have the "on" variant directly iterate over the enemy's unit collection, but then I was lazy when I remembered transported units exist (at -1,-1).

You could probably do it, but it kinda makes sense to leave them separate IMO.